### PR TITLE
Remove be and parse

### PR DESCRIPTION
**Pull request by prestwich**
Remove the least useful functions from the on-chain code. 

Includes #124

### CODE REVIEW

File: c/csrc/btcspv.c [hunk 1]
```diff
@@ -183,11 +183,6 @@ byte_view_t btcspv_extract_input_tx_id_le(const_view_t *tx_in) {
   return tx_id_le;
 }
 
-void btcspv_extract_input_tx_id_be(uint256 hash, const_view_t *tx_in) {
-  const_view_t le = btcspv_extract_input_tx_id_le(tx_in);
-  buf_rev(hash, le.loc, le.len);
-}
-
```
Comment (ErinHales) [inline c/csrc/btcspv.c]: Remove in `btcspv.h`

File: c/csrc/btcspv.c [hunk 2]
```diff
@@ -361,11 +356,6 @@ byte_view_t btcspv_extract_merkle_root_le(const_view_t *header) {
   return root;
 }
 
-void btcspv_extract_merkle_root_be(uint256 hash, const_view_t *header) {
-  const_view_t le = btcspv_extract_merkle_root_le(header);
-  buf_rev(hash, le.loc, le.len);
-}
-
```
Comment (ErinHales) [inline c/csrc/btcspv.c]: Remove in `btcspv.h`

File: c/csrc/btcspv.c [hunk 3]
```diff
@@ -393,11 +383,6 @@ byte_view_t btcspv_extract_prev_block_hash_le(const_view_t *header) {
   return prev_hash;
 }
 
-void btcspv_extract_prev_block_hash_be(uint256 hash, const_view_t *header) {
-  const_view_t le = btcspv_extract_prev_block_hash_le(header);
-  buf_rev(hash, le.loc, le.len);
-}
-
```
Comment (ErinHales) [inline c/csrc/btcspv.c]: Remove in `btcspv.h`

File: golang/cli/header.go [hunk 4]
```diff
@@ -44,7 +44,7 @@ func prettifyHeaderData(
 // ParseHeader takes in a header and returns information about that header: digest, version, previous header hash, merkle root, timestamp, target and nonce
```
Comment (ErinHales) [inline golang/cli/header.go]: Create `header_test.go` and move in related tests.

File: golang/cli/vin.go [hunk 5]
```diff
@@ -1,17 +1,31 @@
 package main
```
Comment (ErinHales) [inline golang/cli/vin.go]: Create `vin_test.go` and move in related tests.

File: golang/cli/vout.go [hunk 6]
```diff
@@ -1,22 +1,37 @@
 package main
```
Comment (ErinHales) [inline golang/cli/vout.go]: Create `vout_test.go` and move in related tests.

File: testVectors.json [hunk 7]
```diff
@@ -579,128 +573,6 @@
       "output": "0x48e5a1a0e616d8fd92b4ef228c424e0c816799a256c6a90892195ccfc53300d6"
     }
   ],
-  "parseInput": [
-    {
```
Comment (ErinHales) [inline testVectors.json]: Are we sure we want to delete these test vectors?  It looks like you moved `parseInput` inside `/golang/cli/vin.go` and `parseOutput` inside `/golang/cli/vout.go`.  If those functions are still present somewhere, we should keep these vectors and test them.

File: golang/cli/header.go (hunk 4 above)
Comment (prestwich) [reply golang/cli/header.go]: these are now officially untested "example" code :)

File: golang/cli/vin.go (hunk 5 above)
Comment (prestwich) [reply golang/cli/vin.go]: these are now officially untested "example" code :)

File: testVectors.json (hunk 7 above)
Comment (prestwich) [reply testVectors.json]: Yes definitely. The moved code is now officially untested "example" code. That means we don't have to maintain tests or quality. #deletedcodeisdebuggedcode :)

**Pull request merged**